src/neural/network_blend.cc

```diff
@@ -0,0 +1,169 @@
+/*
+ This file is part of Leela Chess Zero.
+ Copyright (C) 2018-2021 The LCZero Authors
+
+ Leela Chess is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Leela Chess is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+
+  Additional permission under GNU GPL version 3 section 7
+
+  If you modify this Program, or any covered work, by linking or
+  combining it with NVIDIA Corporation's libraries from the NVIDIA CUDA
+  Toolkit and the NVIDIA CUDA Deep Neural Network library (or a
+  modified version of those libraries), containing parts covered by the
+  terms of the respective license agreement, the licensors of this
+  Program grant you additional permission to convey the resulting work.
+ */
+
+#include "neural/factory.h"
+#include "neural/network.h"
+#include "utils/logging.h"
+
+namespace lczero {
+
+namespace {
+
+class BlendNetwork;
+
+class BlendComputation : public NetworkComputation {
+ public:
+  BlendComputation(std::unique_ptr<NetworkComputation> work_comp,
+                   std::unique_ptr<NetworkComputation> policy_comp)
+      : work_comp_(std::move(work_comp)),
+        policy_comp_(std::move(policy_comp)) {}
+
+  void AddInput(InputPlanes&& input) override {
+    InputPlanes x = input;
+    InputPlanes y = input;
+    work_comp_->AddInput(std::move(x));
+    policy_comp_->AddInput(std::move(y));
+  }
+
+  void ComputeBlocking() override {
+    work_comp_->ComputeBlocking();
+    policy_comp_->ComputeBlocking();
+  }
+
+  int GetBatchSize() const override {
+    return static_cast<int>(work_comp_->GetBatchSize());
+  }
+
+  float GetQVal(int sample) const override {
+    return work_comp_->GetQVal(sample);
+  }
+
+  float GetDVal(int sample) const override {
+    return work_comp_->GetDVal(sample);
+  }
+
+  float GetMVal(int sample) const override {
+    return work_comp_->GetMVal(sample);
+  }
+
+  float GetPVal(int sample, int move_id) const override {
+    return policy_comp_->GetPVal(sample, move_id);
+  }
+
+ private:
+  std::unique_ptr<NetworkComputation> work_comp_;
+  std::unique_ptr<NetworkComputation> policy_comp_;
+};
+
+class BlendNetwork : public Network {
```

Comment:
(probably too late to change)
I'm not sure if `Blend` is the right name for it. From the name I guessed that it's like the "network_switch", but evaluates both nets and based on number of pieces computes weighted average of outputs from two nets.

Reply:
Not late at all, this wasn't supposed to be in this PR at all...